### PR TITLE
[flutter_tools] Add bot configuration to run web_tool_tests for linux, mac, and windows

### PR DESCRIPTION
**Pull request by annagrin**
## Description

Add bot configuration to run web_tool_tests for linux and windows. 

## Related Issues

#69711
#69914

## Tests

The change is only in test and test configuration.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. 

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 1]
```diff
@@ -536,7 +536,10 @@ class LocalEngineArtifacts implements Artifacts {
   @override
   String getArtifactPath(Artifact artifact, { TargetPlatform platform, BuildMode mode }) {
     platform ??= _currentHostPlatform(_platform);
-    final String artifactFileName = _artifactToFileName(artifact, platform, mode);
+    final String artifactFileName =
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/artifacts.dart:539]: nit: don't split the lines here

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: dev/prod_builders.json [hunk 2]
```diff
@@ -329,6 +335,12 @@
          "repo": "flutter",
          "task_name": "win_tool_tests",
          "flaky": false
+      },
+      {
+         "name": "Windows web_tool_tests",
+         "repo": "flutter",
+         "task_name": "win_web_tool_tests",
+         "flaky": true
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply dev/prod_builders.json:343]: No, this property only affects how the result is visualized in the flutter-dashboard and whether it is blocking or not. For presubmit there is a *_try.json which supports an enabled property to disable the build.

[the remark this answers is not in the document]
Comment (jonahwilliams) [reply dev/prod_builders.json:343]: Ahh, so we should probably disable this in presubmit and then leave flaky in post?

[the remark this answers is not in the document]
Comment (godofredoc) [reply dev/prod_builders.json:343]: Yes, although I'd recommend to wait to submit this change until the builds pass in the milo ui:

https://ci.chromium.org/p/flutter/g/framework/console

[the remark this answers is not in the document]
Comment (annagrin) [reply dev/prod_builders.json:343]: Ah, the fix to the failure is actually in this PR:) I can pull it out to do a proper two step fix if needed.

**Review by jonahwilliams — COMMENTED** [review]
Either is fine with me :)

File: dev/prod_builders.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (godofredoc) [reply dev/prod_builders.json:343]: If the fix is already in the PR let's land it.

**Review by godofredoc — APPROVED** [review]

File: dev/prod_builders.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (annagrin) [reply dev/prod_builders.json:343]: @jonahwilliams I would like to see the bots passing first. 

Disabled the failing expression evaluation test in: https://github.com/flutter/flutter/pull/70485

**Comment by annagrin** [comment]
Note: the push after approvals is rebase-only to pick up disabled web_tool_tests.

**Comment by annagrin** [comment]
Disabled windows web_tool_tests due to flakiness. Added mac web_tool_tests (disabled) so we are ready to switch them on after https://flutter-review.googlesource.com/c/recipes/+/8646 lands.

**Review by jonahwilliams — APPROVED** [review]
Still LGTM

**Pull request merged**